File: palindrome.js
{
	let str1;
	let n,i; 	
	let p;
	chkPalindrome("mom");
	chkPalindrome("101");
	chkPalindrome("malayalam");
	chkPalindrome("asd");
	chkPalindrome("a");
}

function chkPalindrome(str1)
{
	let x,i,j;
	x = str1.length-1;
	let str2='';
	for(let i=x;i>=0;i--)
    {
        //console.log(i);
        str2+=str1[i];
    }
	if(str1===str2)
	{
		let flag=1;
		testcase(flag,1)
	}
	else
	{
		let flag=0;
		testcase(flag,0);
	}
}
function testcase(result,Expectedresult)
{
	if(result==Expectedresult)
	{
		console.log("Test successful");
	}
	else
	{
		console.log("Not Successful");
	}
}

File: palindrome.c
#include<stdio.h>
#include<string.h>
void chkPalindrome(char str1[]);
void testcase(int result,int Expectedresult);
int main()
{
	char str1[100];
	int n,i; 	
	int p;
	chkPalindrome("mom");
	chkPalindrome("101");
	chkPalindrome("malayalam");
	chkPalindrome("asd");
	chkPalindrome("a");
}

void chkPalindrome(char str1[])
{
	int x,i,j;
	x = strlen(str1);
	char str2[x];
	strcpy(str2,str1);
	strrev(str2);
	if(strcmp(str1,str2)==0)
	{
		int flag=1;
		testcase(flag,1);
	}
	else
	{
			int flag=0;
		testcase(flag,0);
	}
	
}
void testcase(int result,int Expectedresult)
{
	if(result==Expectedresult)
	{
		printf("\nTest successful");
	}
	else
	{
		printf("\nNot Successful");
	}
}
